### PR TITLE
Use correct interface types in linking

### PR DESCRIPTION
**Pull request by joshuahannan**
The public links need to use all the types in `NonFungibleToken` and `TopShot` for these purposes so they are compatible with as many 3rd party apps as possible. The Top Shot team should make sure they're following this transaction's model for linking an account

### CODE REVIEW

**Review by jrkhan — APPROVED** [review]

**Comment by jrkhan** [comment]
Out of curiosity, could this transaction also be used to relink account collections?
Per @bjartek request:
https://github.com/dapperlabs/nba-smart-contracts/issues/204

**Comment by bjartek** [comment]
Not the way it is written right now no.

File: transactions/shardedCollection/setup_sharded_collection.cdc [hunk 1]
```diff
@@ -25,7 +26,7 @@ transaction(numBuckets: UInt64) {
                 acct.unlink(/public/MomentCollection)
             }
 
-            acct.link<&{TopShot.MomentCollectionPublic}>(/public/MomentCollection, target: /storage/ShardedMomentCollection)
+            acct.link<&{NonFungibleToken.Receiver, NonFungibleToken.CollectionPublic, TopShot.MomentCollectionPublic}>(/public/MomentCollection, target: /storage/ShardedMomentCollection)
```
Comment (satyamakgec) [inline transactions/shardedCollection/setup_sharded_collection.cdc:29]: Why not `MetadataViews.ResolverCollection` here ?

Comment (joshuahannan) [reply transactions/shardedCollection/setup_sharded_collection.cdc:29]: the sharded collection doesn't implement that interface so we can't link it

**Pull request merged**